### PR TITLE
New Data Source: `azurerm_kubernetes_cluster`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Adding a Data Source for AKS since they can be useful to reference

### CODE REVIEW

File: azurerm/data_source_kubernetes_cluster.go [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+		values["client_secret"] = *secret
+	}
+
+	servicePrincipalProfiles.Add(values)
+
+	return servicePrincipalProfiles
+}
+
+func flattenKubernetesClusterDataSourceAccessProfile(profile *containerservice.ManagedClusterAccessProfile) (*string, []interface{}) {
+	if profile != nil {
```
Comment (katbyte) [inline azurerm/data_source_kubernetes_cluster.go]: Could be inverted to return `return nil, []interface{}{}` if nil and to reduce nesting.

File: azurerm/data_source_kubernetes_cluster.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+	values["ssh_key"] = sshKeys
+
+	return []interface{}{values}
+}
+
+func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservice.AgentPoolProfile) []interface{} {
+	agentPoolProfiles := make([]interface{}, 0)
+
+	if profiles := input; profiles != nil {
```
Comment (katbyte) [inline azurerm/data_source_kubernetes_cluster.go]: check could be inverted to reduce nesting.

File: website/docs/d/kubernetes_cluster.html.markdown [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the managed Kubernetes Cluster
+
+* `resource_group_name` - (Required) The name of the Resource Group in which the managed Kubernetes Cluster exists.
```
Comment (katbyte) [inline website/docs/d/kubernetes_cluster.html.markdown]: Some lines end with a `.`, some do not

Comment (tombuildsstuff) [reply website/docs/d/kubernetes_cluster.html.markdown]: 👍 fixed

File: azurerm/data_source_kubernetes_cluster.go (hunk 2 above)
Comment (tombuildsstuff) [reply azurerm/data_source_kubernetes_cluster.go]: 👍 fixed

File: azurerm/data_source_kubernetes_cluster.go (hunk 1 above)
Comment (tombuildsstuff) [reply azurerm/data_source_kubernetes_cluster.go]: 👍 fixed

File: azurerm/data_source_kubernetes_cluster.go [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+		Read: dataSourceArmKubernetesClusterRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
```
[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply azurerm/data_source_kubernetes_cluster.go]: 👍 fixed

**Comment by tombuildsstuff** [comment]
Tests pass:

screen shot 2018-05-07 at 20 03 39

**Pull request merged**